src/api.js

```diff
@@ -23,50 +23,52 @@ module.exports = function(ctx) {
     getSelectedIds: function () {
       return ctx.store.getSelectedIds();
     },
-    add: function (geojson, validateGeoJSON=true) {
-       if (geojson.type !== 'FeatureCollection' && !geojson.geometry) {
-        geojson = {
-          type: 'Feature',
-          id: geojson.id,
-          properties: geojson.properties || {},
-          geometry: geojson
-        };
+    set: function(featureCollection) {
+      if (featureCollection.type === undefined || featureCollection.type !== 'FeatureCollection' || !Array.isArray(featureCollection.features)) {
+        throw new Error('Invalid FeatureCollection');
       }
+      var toDelete = JSON.parse(JSON.stringify(ctx.store.getAllIds()));
```

Comment:
Could clone this with a slice, right? `ctx.store.getAllIds().slice()` Maybe not ...